client/lib/layout-focus/index.js

```
@@ -1,9 +1,51 @@
+/** @ssr-ready **/
+
 /**
  * Internal dependencies
  */
-var Emitter = require( 'lib/mixins/emitter' ),
-	config = require( 'config' );
+import Emitter from 'lib/mixins/emitter';
+import config from 'config';
+import { getCurrentLayoutFocus, getPreviousLayoutFocus } from 'state/ui/selectors';
+import { setLayoutFocus, setNextLayoutFocus, activateNextLayoutFocus } from 'state/ui/actions';
+
+// These are the structural areas
+// of the main body of Calypso
+const _areas = [ 'content', 'sidebar', 'sites', 'preview' ];
+function isValid( area ) {
+	const valid = _areas.indexOf( area ) !== -1;
+
+	if ( ! valid ) {
+		if ( config( 'env' ) === 'development' ) {
+			throw new Error( area + ' is not a valid layout focus area' );
+		}
+	}
 
+	return valid;
+}
+
+// This is unfortunately necessary to avoid layout elements still in the DOM
+// being targeted by things like browser inline search, which may cause odd
+// positioning effects by having their CSS modified.
+function setFocusHideClass() {
+	if ( typeof document === 'undefined' ) {
+		return;
+	}
+	// Whenever layout focus changes remove `focus-hide` so
+	// that animations can occur with all elements visible.
+	document.documentElement.classList.remove( 'focus-hide' );
+
+	// After transitions restore `focus-hide`
+	setTimeout( function() {
+		document.documentElement.classList.add( 'focus-hide' );
+	}, 200 );
+}
+
+// Used to keep a reference to the Redux store so we can update it when the
+// legacy Flux store is updated.
+let reduxStore;
```

@aduth I was thinking that we could remove the legacy layoutFocus code once we converted some of the early uses. 

On the other hand, I think it won't be used by anything, so instead we could just throw an error if something tries to use `layoutFocus` before it's bound to the Redux store now. That actually might be a quick way to find anything that might do that.

That said, I was hoping to make this PR relatively risk-free, and if we introduce a possible new error, that feels less safe to me. What do you think?